test/functional/autocmd/dirchanged_spec.lua

```diff
@@ -0,0 +1,52 @@
+local lfs = require('lfs')
+local h = require('test.functional.helpers')(after_each)
+
+local clear = h.clear
+local eq = h.eq
+local eval = h.eval
+local execute = h.execute
+
+describe('Event DirChanged ->', function()
+  local wd = lfs.currentdir()
+  local dirs = {
+    wd .. '/Xtest-functional-autocmd-dirchanged.dir1',
+    wd .. '/Xtest-functional-autocmd-dirchanged.dir2',
+    wd .. '/Xtest-functional-autocmd-dirchanged.dir3',
+  }
+
+  setup(function()    for _, dir in pairs(dirs) do h.mkdir(dir) end end)
+  teardown(function() for _, dir in pairs(dirs) do h.rmdir(dir) end end)
+
+  before_each(clear)
+
+  describe('"autocmd DirChanged', function()
+    it('*" sets v:event for all :cd variants', function()
+      execute('autocmd DirChanged * let event = copy(v:event)')
+
+      execute('lcd '..dirs[1])
+      eq({path=dirs[1], scope='window'}, eval('event'))
+
+      execute('tcd '..dirs[2])
+      eq({path=dirs[2], scope='tab'}, eval('event'))
+
+      execute('cd '..dirs[3])
+      eq({path=dirs[3], scope='global'}, eval('event'))
+    end)
+
+    -- TODO(mhinz): Add more tests here if we start to make
+    --              use of the autocmd pattern.
```

Comment:
To test the API, could use `request("nvim_set_current_dir")`, e.g. https://github.com/justinmk/neovim/blob/a85d116b1c4360da05ef1d6bc75e322b0b0b8e0d/test/functional/autocmd/bufenter_spec.lua#L16